bundles/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/internal/AutelisGenericBindingProvider.java

```diff
@@ -57,7 +57,7 @@ public String getBindingType() {
      */
     @Override
     public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
-        if (!(item instanceof SwitchItem || item instanceof DimmerItem || item instanceof NumberItem
+        if (!(item instanceof SwitchItem || item instanceof NumberItem
```

Comment:
Please explain the removal of the `DimmerItem` checks throughout...

Reply:
`DimmerItem` is a subclass of `SwitchItem`. In the above code (and in other places) the test `item instanceof DimmerItem` is being made after it has been established that `item` is not an instance of `SwitchItem` (and therefore not an instance of any subclass of `SwitchItem`, including `DimmerItem`). So in this case it's a redundant check. In some other cases, there is also dead code following the check.